example/main.go

```diff
@@ -21,17 +23,21 @@ func main() {
 		flag.Usage()
 		return
 	}
-	db, err := notmuch.Open(*dir, notmuch.DB_RDONLY)
+	db, err := notmuch.Open(*dir, notmuch.DBReadOnly)
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
 	defer db.Close()
-	threads, err := db.QueryCreate(*queryString).SearchThreads()
+	threads, err := db.NewQuery(*queryString).Threads()
 	if err != nil {
 		fmt.Println(err)
 	}
-	for ; threads.Valid(); threads.MoveToNext() {
+	for {
+		if err := threads.MoveToNext(); err != nil {
+			break
+		}
+
```

Comment:
That's a good point, actually in my tests I have it running at the end of the result. I'll add a test that panics and work from there.